modules/dxp/apps/search-experiences/search-experiences-blueprints-admin-web/src/main/resources/META-INF/resources/js/edit_blueprint/clause_contributors_tab/index.js

```diff
@@ -151,6 +175,15 @@ function ClauseContributorsTab({initialContributors}) {
 		setContributors(newContributors);
 	}, [enabled, category, keyword, sortDirection, status]); //eslint-disable-line
 
+	useEffect(() => {
+		onFrameworkConfigClauseChange(
+			getFrameworkConfigClauseContributors(
+				initialClauseContributorsList,
+				enabled
+			)
+		);
+	}, [enabled]); //eslint-disable-line
```

Comment:
I'd be really hesitant disabling the eslint here and the useEffect before as well. Why are we not able to include `onFrameworkConfigClauseChange` and `initialClauseContributorsList`? It can lead to stale data being referenced which is why the eslint rule exists in the first place.

Reply:
If it's because of `onFrameworkConfigClauseChange` changing too often, we can wrap the function in a `useCallback` before passing it down. We should be avoiding inline functions too since they cause components to re-render. For example we should avoid:

```javascript
<SomeComponent onChange={() => {console.log("I am in inline function");}} />
```

Reply:
I'm not sure what happened, but my browser seemed to crash when I added either `onFrameworkConfigClauseChange` or `initialClauseContributorsList`, even when I used useCallback for the `onFrameworkConfigClauseChange`. I think it still kept rerendering? In the commit I added, I just ended up changing it so that it doesn't rely on useEffect anymore, it just applies updates alongside `enabled`. The onChange function also worked out to be reusable for the query builder. Does that work? Let me know if I should change it 😅

Reply:
Does this only happen in storybook or in portal as well? I'm still not sure exactly how storybook handles the props passing down, but sometimes I would have issues with the props not properly applying.

Reply:
Oh good question... I only tried it on storybook 😬